dbms/src/Server/Client.cpp

```diff
@@ -197,7 +197,20 @@ class Client : public Poco::Util::Application
 
             default_database = config.getString("database", "");
             user = config.getString("user", "");
-            password = config.getString("password", "");
+
+            if (config.getBool("ask-password", false))
+            {
+                std::string prompt = "Password for user " + user + ": ";
+                if (char * line_ = getpass(prompt.c_str()))
+                {
+                    password = line_;
+                    free(line_);
+                }
+            }
+            else
+            {
+                password = config.getString("password", "");
```

Comment:
Missing check for the case when both password and ask-password options was specified in command line.